modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/IdentityProxy.java

```diff
@@ -110,7 +110,7 @@ public void init(CredentialInfo info, APIAccessCallBack apiAccessCallBack, Conte
 		accessTokenHandler.obtainAccessToken();
 	}
 
-	public void requestToken(Context contextt, TokenCallBack call, String clientID,
+	public synchronized void requestToken(Context contextt, TokenCallBack call, String clientID,
```

Comment:
Without synchronizing the whole method, let's try to synchronize the most relevant segments to avoid any overhead introduced by the locks.